File: payment-network/src/main/java/com/payment/agent/transaction/service/TransactionService.java
package com.payment.agent.transaction.service;

import com.payment.agent.transaction.dtos.TransactionRequest;
import com.payment.agent.transaction.dtos.TransactionResponse;

import java.util.List;

public interface TransactionService {

    /**
     * Create a new transaction locally
     * @param request Transaction details
     * @return Created transaction with PENDING_SYNC status
     */
    TransactionResponse createTransaction(TransactionRequest request);

    /**
     * Get all local transactions
     * @return List of transactions
     */
    List<TransactionResponse> getLocalTransactions();
}

File: agent-interface/src/features/transaction/services/transactionService.ts
import axios from 'axios';
import { Transaction, TransactionRequest, TransactionResponse } from '../types/transaction.types';

// Base URL for agent API - can be configured via env variable
const API_BASE_URL = process.env.REACT_APP_AGENT_API_URL || 'http://localhost:8081';

const api = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});

// Create a new transaction (deposit or withdrawal)
export const createTransaction = async (request: TransactionRequest): Promise<TransactionResponse> => {
  const response = await api.post<TransactionResponse>('/api/agent/transactions', request);
  return response.data;
};

// Get all local transactions from this agent
export const getLocalTransactions = async (): Promise<Transaction[]> => {
  const response = await api.get<Transaction[]>('/api/agent/transactions');
  return response.data;
};
